pilot/pkg/xds/auth.go

```diff
@@ -56,17 +56,20 @@ func (s *DiscoveryServer) authenticate(ctx context.Context) ([]string, error) {
 	if _, ok := peerInfo.AuthInfo.(credentials.TLSInfo); !ok && !AuthPlaintext {
 		return nil, nil
 	}
-	authFailMsgs := []string{}
+
+	authContext := security.NewAuthContext(ctx)
+
 	for _, authn := range s.Authenticators {
-		u, err := authn.Authenticate(ctx)
+		// TODO: Refactor to remove the return value of Authencicate. Just rely on AuthContext.
+		u, err := authn.Authenticate(authContext)
 		// If one authenticator passes, return
 		if u != nil && u.Identities != nil && err == nil {
+			authContext.Caller = u
```

Comment:
Caller is also not used. If we also make the `Messages` change I suggested, we no longer need `AuthContext` entirely anymore